### PR TITLE
chore(async): Making create app configurable

### PR DESCRIPTION
**Pull request by craig-rueda**
### SUMMARY
First of several PRs to revive Global Async Queries.

* Makes AsyncQueryManager implementation configurable
* Adds nginx container to local docker-compose in order to deal with local websocket connections
* Dries up loading of class names to class instances
* Adds optional kwargs to `create_app()` in order to facilitate integration test configs

### CODE REVIEW

File: docker/nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,127 @@
+#
```
Comment (craig-rueda) [inline docker/nginx/nginx.conf:1]: Added this config for local testing

File: superset/app.py [hunk 2]
```diff
@@ -25,12 +26,14 @@
 logger = logging.getLogger(__name__)
 
 
-def create_app() -> Flask:
+def create_app(superset_config_module: Optional[str] = None) -> Flask:
```
Comment (craig-rueda) [inline superset/app.py:29]: Adding optional override module for test configs

File: superset/extensions/ssh.py [hunk 3]
```diff
@@ -78,18 +78,9 @@ def __init__(self) -> None:
...
-        ]
-        ssh_manager_module_name = ssh_manager_fqclass[
-            0 : ssh_manager_fqclass.rfind(".")
-        ]
-        ssh_manager_class = getattr(
-            importlib.import_module(ssh_manager_module_name), ssh_manager_classname
-        )
-
-        self._ssh_manager = ssh_manager_class(app)
+        self._ssh_manager = load_class_from_name(
```
Comment (craig-rueda) [inline superset/extensions/ssh.py:81]: Drying this up

File: superset/utils/machine_auth.py [hunk 4]
```diff
@@ -100,18 +100,9 @@ def __init__(self) -> None:
...
-        ]
-        auth_provider_module_name = auth_provider_fqclass[
-            0 : auth_provider_fqclass.rfind(".")
-        ]
-        auth_provider_class = getattr(
-            importlib.import_module(auth_provider_module_name), auth_provider_classname
-        )
-
-        self._auth_provider = auth_provider_class(app.config["WEBDRIVER_AUTH_FUNC"])
+        self._auth_provider = load_class_from_name(
```
Comment (craig-rueda) [inline superset/utils/machine_auth.py:103]: Drying up

File: superset/async_events/async_query_manager_factory.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    def __init__(self) -> None:
+        self.async_query_manager: AsyncQueryManager = None  # type: ignore
+
+    def init_app(self, app: Flask) -> None:
+        self.async_query_manager = load_class_from_name(
+            app.config["GLOBAL_ASYNC_QUERY_MANAGER_CLASS"]
+        )()
+        self.async_query_manager.init_app(app)
+
+    def get(self) -> AsyncQueryManager:
```
Comment (dpgaspar) [inline superset/async_events/async_query_manager_factory.py:34]: nit: on `MachineAuthProviderFactory` we use `def instance(self) -> MachineAuthProvider:` and the property that holds the instance it self is "private"

**Review by dpgaspar — APPROVED** [review]

**Review by hughhhh — APPROVED** [review]

**Pull request merged**